modules/Rock.cmake

```diff
@@ -765,28 +759,85 @@ endfunction()
 # files, they get added to the library and the corresponding header file is
 # passed to moc.
 function(rock_testsuite TARGET_NAME)
+    rock_test_common(${TARGET_NAME} ${ARGN})
+    rock_setup_boost_test()
+    rock_add_test()
+endfunction()
+
+## Uses gtest + google-mock as unit testing framework
+# The interface is the same as the one from rock_testsuite
+#
+# Unfortunately, gtest and google-mock do not ship with
+# pre-compiled libraries so we have to add their source files
+# to the test target. If GTEST_DIR or GMOCK_DIR are not set,
+# this function will look for the required source files
+# in /usr/src, which is the default path under ubuntu/debian
+function(rock_gtest TARGET_NAME)
+    if (NOT GTEST_DIR)
+        find_path(GTEST_DIR "src/gtest-all.cc" /usr/src/gtest)
+    endif()
+    if (NOT GMOCK_DIR)
+        find_path(GMOCK_DIR "src/gmock-all.cc" /usr/src/gmock)
+    endif()
+
+    if (NOT GTEST_DIR)
+        message(FATAL_ERROR "Could not find gtest in /usr/src")
+    endif()
+    if (NOT GMOCK_DIR)
+        message(FATAL_ERROR "Could not find google-mock in /usr/src")
+    endif()
+    message(STATUS "gtest found ... building the test suite")
+    
+    list(APPEND ${TARGET_NAME}_SOURCES ${GTEST_DIR}/src/gtest-all.cc
```

Comment:
Not sure I got it right.. Do you mean passing the sources to rock_test_common and then let it pass to rock_executable? My concern was mixing internal ARGN and user's supplied ARGN in rock_executable.. i.e the user does: 

```
rock_gtest(foo_test SOURCES foo_src.cpp)
```

We would internally do:

```
rock_executable(foo_test /usr/src/gtest/src/gtest-al.cc /usr/src/gmock-all.cc SOURCES foo_src.cpp)
```

Does that work? By appending the sources to ${TARGET_NAME}_SOURCES we are sure that these files are being added as sources...